src/containers/ToDoList/ToDoList.jsx

```diff
@@ -71,9 +73,12 @@ export const ToDoList = () => {
   }, []);
 
   const handleTaskMark = (id) => {
-    const newTasksList = tasks.map((task) =>
+    const updatedTasks = tasks.map((task) =>
       task.id === id ? { ...task, checked: !task.checked } : task
     );
+
+    const newTasksList = updatedTasks.sort((a, b) => a.checked - b.checked);
```

Comment:
Operations like filtering and/or sorting should be performed while reading the state, not while writing.